keps/sig-auth/4601-authorize-with-selectors/README.md

```diff
@@ -0,0 +1,819 @@
+selector has no semantic meaning.
+In practice this means that (for now), only List, Watch, and DeleteCollection have field and label selectors.
+SubjectAccessReviews submitted to the kube-apiserver with verbs that do not honor the selectors will have the field and
+label selector attributes removed prior to checking authorization.
+Webhook authors: remember that the list of verbs accepting field and label selectors may change over time.
+If the kube-apiserver sends the FieldSelector or LabelSelector to a webhook, the kube-apiserver intends to honor the selector attributes.
+Any authorizer that gets an error from `ParseFieldSelector` or `ParseLabelSelector` may attempt to authorize without
+field or label selectors since that will authorize using a wider permission (field and label selectors can only reduce access).
+
+```go
+type Attributes interface {
+	// ParseLabelSelector is lazy, thread-safe, and stores the parsed result and error.
+	// It can return an error if the field selector cannot be parsed.
+    // Remember that field selector formats vary based on the version of the API being used!
+	ParseFieldSelector() ([]Requirement, error)
+  
+	// ParseLabelSelector is lazy, thread-safe, and stores the parsed result and error.
+    // It can return an error if the field selector cannot be parsed.
+    ParseLabelSelector() ([]Requirement, error)
+```
+
+#### Future-proofing your authentication webhook for future verbs
+
+As of 1.31, the only verbs with field and label selectors are List, Watch, and DeleteCollection.
+<<[UNRESOLVED deads2k, liggitt, jpbetz, sttts: this is a future promise for the kube-apiserver behavior if we add it ]>>
+In the future, the kube-apiserver may add field and label selectors to Create, Update, and Delete.
+<<[/UNRESOLVED]>>
+* For Create, this means that the resource after all mutation is complete (finalObject) must match the field and label selector. 
+* For Update, this means that the finalNewObject and oldObject must match the field and label selector.
+* For Delete, this means that the oldObject must match the field and label selector.
+ 
+We do not expand field and label selectors for Get, because if a client is specifying a selector, they can add a `.metadata.name`
+field selector and use a List to get equivalent functionality.
+
+### SubjectAccessReview Changes
+
+SubjectAccessReview is used for two purposes:
+1. Authorization webhook calls from the kube-apiserver to a webhook.
+   This usage likely benefits from a serialization with `[]Requirement`.
+2. Authorization checks from a client (often a server process using in-cluster authorization like kube-rbac-proxy)
+   This usage likely benefits from a serialization that matches the query parameter.
+
+Their needs best met with two different serialization (see user stories)
+
+```go
+type SubjectAccessReviewSpec struct {
+	FieldSelector *FieldSelectorAttributes
+	
+	LabelSelector *LabelSelectorAttributes
```

Comment:
would these go under `ResourceAttributes` rather than directly under the `SubjectAccessReviewSpec`?